pkg/manifests/manifests.go

```diff
@@ -1358,9 +1362,156 @@ func (f *Factory) PrometheusK8s(host string, grpcTLS *v1.Secret, trustedCABundle
 		}
 	}
 
+	if f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.AlertmanagerConfigs != nil {
+		p.Spec.AdditionalAlertManagerConfigs = &v1.SecretKeySelector{
+			Key: AdditionalAlertmanagerConfigSecretKey,
+			LocalObjectReference: v1.LocalObjectReference{
+				Name: AdditionalAlertmanagerConfigSecretName,
+			},
+		}
+		p.Spec.Secrets = append(p.Spec.Secrets, getAlertmanagerTLSSecretsName(f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.AlertmanagerConfigs)...)
+	}
+
 	return p, nil
 }
 
+func (f *Factory) AdditionalAlertManagerConfigsSecret() (*v1.Secret, error) {
+	alertmanagerConfigs := f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.AlertmanagerConfigs
+	if len(alertmanagerConfigs) == 0 {
+		return nil, nil
+	}
+
+	cfgs := []yaml2.MapSlice{}
+	for _, alertmanagerConfig := range alertmanagerConfigs {
+		cfg := yaml2.MapSlice{}
+
+		if alertmanagerConfig.Scheme != "" {
+			cfg = append(cfg, yaml2.MapItem{
+				Key: "scheme", Value: alertmanagerConfig.Scheme,
+			})
+		}
+		if alertmanagerConfig.PathPrefix != "" {
+			cfg = append(cfg, yaml2.MapItem{
+				Key: "path_prefix", Value: alertmanagerConfig.PathPrefix,
+			})
+		}
+		if alertmanagerConfig.APIVersion != "" {
+			cfg = append(cfg, yaml2.MapItem{
+				Key: "api_version", Value: alertmanagerConfig.APIVersion,
+			})
+		}
+		if alertmanagerConfig.Timeout != nil {
+			cfg = append(cfg, yaml2.MapItem{
+				Key: "timeout", Value: alertmanagerConfig.Timeout,
+			})
+		}
+
+		auth := yaml2.MapSlice{}
+		if alertmanagerConfig.BearerToken != nil {
+			if alertmanagerConfig.BearerToken.Name == "" {
+				return nil, errors.Errorf("bearer token secret name not found")
+			}
+			if alertmanagerConfig.BearerToken.Key == "" {
+				return nil, errors.Errorf("bearer token secret key not found")
+			}
+			auth = append(auth, yaml2.MapItem{
+				Key: "credentials_file", Value: fmt.Sprintf("/etc/prometheus/secrets/%s/%s",
+					alertmanagerConfig.BearerToken.Name, alertmanagerConfig.BearerToken.Key),
+			})
+		}
+		if len(auth) != 0 {
+			cfg = append(cfg, yaml2.MapItem{
+				Key:   "authorization",
+				Value: auth,
+			})
+		}
+
+		tlsConfig := yaml2.MapSlice{}
+		if alertmanagerConfig.TLSConfig.CA != nil {
+			if alertmanagerConfig.TLSConfig.CA.Name == "" {
+				return nil, errors.Errorf("ca secret name not found")
```

Comment:
same remark about adding the secret name in the error message (also valid for all other returns)